### PR TITLE
Allow use of the Mobilefrontend content provider proxy

### PR DESCRIPTION
**Pull request by jdlrobson**
Untested, but I'd like to add a proxy checkbox for testing languages as part of the desktop improvements project. It's really hard to test these without and I've been having to workaround this by using https://gerrit.wikimedia.org/r/c/mediawiki/extensions/MobileFrontend/+/641808

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -146,6 +146,19 @@
...
 					),
+					new OOUI\FieldLayout(
+						new OOUI\CheckboxInputWidget( [
+							'name' => 'proxy',
+							'value' => 1,
+							'selected' => false
+						] ),
+						[
+							'label' => 'Proxy articles from en.wikipedia.org',
+							'help' => 'Any content not locally to the wiki will be pulled from production wiki	',
```
Comment (edg2s) [inline index.php:157]: content->articles
locally->local

or:
"Any articles which don't exist on the wiki will be fetched from English Wikipedia"

**Comment by edg2s** [comment]
This seems generally useful enough to me. We may need to consider an "advanced options" section in the near future.

File: index.php [hunk 2]
```diff
@@ -146,6 +146,19 @@
 							'classes' => [ 'form-repos-field' ],
 						]
 					),
+					new OOUI\FieldLayout(
+						new OOUI\CheckboxInputWidget( [
+							'name' => 'proxy',
+							'value' => 1,
+							'selected' => false
```
Comment (edg2s) [inline index.php:153]: Should probably be disabled by default.

Comment (jdlrobson) [reply index.php:153]: Disabled or checked (selected)?

Comment (edg2s) [reply index.php:153]: Sorry, unchecked (selected=false)

Comment (jdlrobson) [reply index.php:153]: Sorry.. still a little confused. what do you want me to change here? You want me to make it selected = true ?

File: index.php (hunk 1 above)
Comment (jdlrobson) [reply index.php]: (done)

**Comment by edg2s** [comment]
If the purpose is to test content rendering it would probably be useful to also import en:MediaWiki:Common.css?

One could either fetch the latest version in PHP, or just add a getCSS call to Common.js (provided there are no CORS issues).

File: createwiki.sh [hunk 3]
```diff
@@ -48,6 +48,11 @@ php $PATCHDEMO/wikis/$NAME/w/maintenance/install.php \
 # apply our default settings
 cat $PATCHDEMO/LocalSettings.txt >> $PATCHDEMO/wikis/$NAME/w/LocalSettings.php
 
+# Add the proxy if selected
+if [ "${USE_PROXY}" = "1" ] then
+	cat $PATCHDEMO/wikis/$NAME/w/LocalSettings.php $PATCHDEMO/LocalSettings-proxy.txt
```
Comment (MatmaRex) [inline createwiki.sh:53]: Missing `>>`?

File: new.php [hunk 4]
```diff
@@ -236,6 +236,8 @@ function set_progress( int $pc, string $label ) {
 // Choose repositories to enable
 $repos = get_repo_data();
 
+$useProxy = $_POST['proxy'];
```
Comment (MatmaRex) [inline new.php:239]: Add `(int)` or some other validaiton thing

File: createwiki.sh (hunk 3 above)
Comment (edg2s) [reply createwiki.sh:53]: and they're the wrong way around
`cat txt >> php`

Comment (jdlrobson) [reply createwiki.sh]: (done)

File: new.php (hunk 4 above)
Comment (jdlrobson) [reply new.php]: (done)

File: createwiki.sh [hunk 5]
```diff
@@ -47,3 +47,8 @@ php $PATCHDEMO/wikis/$NAME/w/maintenance/install.php \
 
 # apply our default settings
 cat $PATCHDEMO/LocalSettings.txt >> $PATCHDEMO/wikis/$NAME/w/LocalSettings.php
+
+# Add the proxy if selected
+if [ "${USE_PROXY}" = "1" ] then
+	cat $PATCHDEMO/wikis/$NAME/w/LocalSettings.php $PATCHDEMO/LocalSettings-proxy.txt >> $PATCHDEMO/wikis/$NAME/w/LocalSettings.php
```
Comment (edg2s) [inline createwiki.sh:53]: >> means append, so you don't need to repeat the php file. I'll fix in the rebase

File: LocalSettings-proxy.txt [hunk 6]
```diff
@@ -0,0 +1,5 @@
+// Allow proxying content from production
+
+$wgMFContentProviderClass = "MobileFrontend\\ContentProviders\\MwApiContentProvider";
+$wgMFContentProviderTryLocalContentFirst = false;
```
Comment (edg2s) [inline LocalSettings-proxy.txt:4]: This option should be dropped then, given what we say in the form below. Otherwise we would always get remote content.

Comment (jdlrobson) [reply LocalSettings-proxy.txt]: Switched to true

**Comment by edg2s** [comment]
This should also enforce that MobileFrontend is selected in the extensions list.

**Review by edg2s — APPROVED** [review]

**Comment by MatmaRex** [comment]
Wild.

**Pull request merged**